text/0000-bool-to-option.md

```diff
@@ -0,0 +1,106 @@
+- Feature Name: `bool_to_option`
+- Start Date: 2019-09-05
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Add methods to `bool` for converting to an `Option<T>`, given a `t: T`, where `true` maps
+to `Some(t)` and `false` maps to `None`, abstracting the following common pattern in Rust.
+
+```rust
+if some_condition {
+    Some(t)
+} else {
+    None
+}
+```
+
+# Motivation
+[motivation]: #motivation
+
+This is an
+[extremely common pattern](https://sourcegraph.com/search?q=repogroup%3Acrates+%2Felse%5Cs*%7B%5Cs*None%5Cs*%7D%2F+count%3A1000)
+in Rust code, but is quite verbose, taking several lines to achieve something with only two important
+data: the `bool` and the `T`. If we instead collapse the expression on to a single line, the legibility of the code is reduced. In
```

Comment:
typo: "the legibility of the code is reduced" -> "the illegibility of the code is reduced"